feat(curriculum): add Artifact Provenance Auditor workshop (#68860)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-artifact-provenance-auditor/6a57c331ae94fa4631b60870.md

---
id: 6a57c331ae94fa4631b60870
title: Step 17
challengeType: 1
dashedName: step-17
---

# --description--

> Finally, you will build a readable summary for any artifact. Start by handling the missing artifact.

Reviewer: ```suggestion
Finally, build a readable summary for any artifact. Start by handling the case where an artifact is missing.
```

> Create a function named `buildSummary` with a parameter `id`. Look up the artifact with `collection[id]`. If no artifact is found, return the string `"Artifact not found"`.

Reviewer: ```suggestion
Create a `buildSummary` function that takes an `id` parameter. Look up the artifact using `collection[id]`. If the artifact is not found, return the string `"Artifact not found"`.
```

---
id: 6a57c331ae94fa4631b60870
title: Step 17
challengeType: 1
dashedName: step-17
---

# --description--

> Finally, you will build a readable summary for any artifact. Start by handling the missing case.

Reviewer: ```suggestion
Finally, you will build a readable summary for any artifact. Start by handling the missing artifact.
```

---
id: 6a57c331ae94fa4631b60870
title: Step 17
challengeType: 1
dashedName: step-17
---

# --description--

Finally, you will build a readable summary for any artifact. Start by handling the missing case.

The lookup line is provided. Inside `buildSummary`, if no artifact is found, return the string `"Artifact not found"`.

# --hints--

You should have a function named `buildSummary` with a parameter named `id`.

```js
assert.isFunction(buildSummary);
const regex = __helpers.functionRegex('buildSummary', ['id']);

> assert.match(__helpers.removeJSComments(code), regex);

Reviewer: You can replace this with AST helpers. Please refer to the [AST Helpers documentation.](https://contribute.freecodecamp.org/curriculum-help/#typescript-helpers).
